controllers/noteController.js

```diff
@@ -0,0 +1,92 @@
+const noteRepository = require("../repositories/noteRepository");
+
+exports.createNote = async (req, res) => {
+  try {
+    const { title, content } = req.body;
+    const note = await noteRepository.createNote(title, content);
+    res.status(201).json(note);
+  } catch (error) {
+    res.status(400).json({ message: "Error creating note", error });
+  }
+};
+
+exports.getNotes = async (req, res) => {
+  try {
+    const { page = 1, limit = 10 } = req.query;
+    const notes = await noteRepository.getAllNotes(page, limit);
+
+    res.json(notes);
+  } catch (error) {
+    res.status(500).json({ message: "Error fetching notes", error });
+  }
+};
+
+exports.updateNote = async (req, res) => {
+  try {
+    const { title, content } = req.body;
+    const existingNote = await noteRepository.getNoteById(req.params.id);
```

Comment:
You could also destructure `req.params` to extract `id`, similar to what you did on line 26. This would make it more consistent, as `id` is used again on line 43.

Reply:
Done